modules/pam_motd/pam_motd.c

```diff
@@ -215,6 +209,47 @@ static void try_to_display_directories_with_overrides(pam_handle_t *pamh,
     if (dirscans_size_total == 0)
         goto out;
 
+    /* filter out unwanted names, directories, and complement data with lstat() */
+    for (i = 0; i < num_motd_dirs; i++) {
+	unsigned int j;
+	int rc, drop;
+	struct dirent **d;
+	char *fullpath;
+	struct stat s;
+
+	d = dirscans[i];
+	for (j = 0; j < dirscans_sizes[i]; j++) {
+	    drop = 1;
+	    switch(d[j]->d_type) {    /* the filetype determines how to proceed */
+	    case DT_REG:              /* regular files and     */
+	    case DT_LNK:              /* symlinks              */
+		drop = 0;             /* are good.             */
```

Comment:
I think we would better off without `drop` variable at all.
For example, if `drop = 0` is replaced with `continue`, then
there would be no need to assign anything to `drop` and check it afterwards.

Reply:
thanks for that, that is a good idea.